nuxeo-features/nuxeo-search/nuxeo-search-core/src/main/java/org/nuxeo/ecm/platform/search/core/SavedSearchWriter.java

```diff
@@ -88,18 +92,25 @@ protected void writeEntityBody(SavedSearch search, JsonGenerator jg) throws IOEx
             jg.writeStringField(param, search.getNamedParams().get(param));
         }
 
+        SchemaManager schemaManager = Framework.getService(SchemaManager.class);
+        DocumentModel doc = search.getDocument();
+        Set<String> ownSchemas = new HashSet<>(Arrays.asList(doc.getSchemas()));
+        doc.getFacets().stream()
+                .map(facet -> schemaManager.getFacet(facet).getSchemas())
```

Comment:
Indentation is not correct, it should be on code you added on maintenance branch.